### DISCUSSION TITLE
Render custom HTML in grape js editor on load

### DISCUSSION THREAD

**Discussion by joyal-jose** [GrapesJS/grapesjs#5069] [Q&A]
I need to render some custom HTML in the grape js editor on load. The data of HTML and CSS come from databases. How can I render the data on load.

**Comment by dm-siegel** [comment]
After you initialize grapes, you can call editor.setComponents(yourHtml) and editor.setStyle(yourCss).
```typescript
this._editor = grapesjs.init({
      container: '#gjs',
      fromElement: true,
      // Size of the editor
      height: '100%',
      width: 'auto',
})

this._editor.setComponents(yourHtml);
this._editor.setStyle(yourCss);
```
